### PR TITLE
Fix issue with top Nav

### PR DESCRIPTION
**Pull request by weyermannx**
This will finally allow rails 4-5 ama-layout to be merged - it will go on every app (Even the new driver ed ones) except rewards (which is waiting for a security scan so it can be added to the layout and still has a custom layout)

- [x] I have deployed every app with a branch of this gem
- [x] I have checked every app for passing travis

🐘 
* gatekeeper nav wasn't able to render between create/new account

* Put tests back mimic what they were before I started this change
* Test to bring coverage to 100% - the ama_layout_view class is needed to pass the view to the decorator so that partials in the app can be rendered by the gem

Addendum to https://github.com/amaabca/ama_layout/pull/219

parts of VSTS-1582 - https://amaabca.visualstudio.com/dx_technical_debt/_queries?id=1582&triage=false&_a=edit

### PR Review Checklist
To be completed by the person who opens the PR. Use strikethroughs for items that are not applicable in list.
- [x] I have reviewed my own PR to check syntax & logic, removed unnecessary/old code, made sure everything aligns with our style guide and BEM.
- [x] Any applicable version numbers have been updated.
- [x] I've tested on mobile, tablet, and desktop, as well as across all of the browsers we support.
- [x] I've proof-read all text for legibility, proper grammar, punctuation, and capitalization (titlecase).
- [x] I have written a detailed PR message explaining what is being changed and why
- [x] I’ve linked to any relevant VSTS tickets
- [x] I’ve added the appropriate review symbols to my PR - (elephant) for dev review, (art) for design

### CODE REVIEW

**Comment by MathieuGilbert** [comment]
📖

File: lib/ama_layout/ama_layout_view.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module AmaLayout
+  class AmaLayoutView < ActionView::Base
+    attr_accessor :view_data
+
+    def initialize(args)
+      self.view_data = args[:view_data]
+      controller = view_data.try(:controller) || ::ApplicationController.new
+      context = controller.view_paths
+      super(context,{},controller)
```
Comment (MathieuGilbert) [inline lib/ama_layout/ama_layout_view.rb]: Could you add some spaces there?

File: lib/ama_layout/decorators/agent/navigation_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        "Welcome, #{display_name.titleize}#{cash_drawer_name}"
+      end
+
+      def sign_out_link
+        return "" unless user
+        h.render partial: "ama_layout/sign_out_link"
+      end
+
+      def top_nav
+        return '' unless user
```
Comment (MathieuGilbert) [inline lib/ama_layout/decorators/agent/navigation_decorator.rb]: Could you make the quotes (single vs double) consistent in this file?

**Comment by MathieuGilbert** [comment]
💩  Just minor things.

**Comment by weyermannx** [comment]
:wave:

**Comment by MathieuGilbert** [comment]
:elephant: :elephant: :elephant:

**Pull request merged**